prow/config/cache.go

```diff
@@ -114,51 +116,39 @@ func NewInRepoConfigCacheGetter(
 		GitHubOptions:  gitHubOptions,
 		CookieFilePath: cookieFilePath,
 		DryRun:         dryRun,
-		CacheMap:       make(map[string]*InRepoConfigCacheHandler),
 	}, nil
 }
 
-func (irc *InRepoConfigCacheGetter) GetCache(cloneURI, host string) (*InRepoConfigCacheHandler, error) {
+// GetCache returns InRepoConfigCacheHandler based on authentication methods. If
+// GitHub bot token or GitHub app private key is provided then returns handler
+// that works best with GitHub, if http.cookiefile is provided then returns
+// handler that authenticate with cookiefile.
+func (irc *InRepoConfigCacheGetter) GetCache() (*InRepoConfigCacheHandler, error) {
 	// No repo is cloned in getCache, Since this function should happen fast it is safe to lock the whole function.
 	irc.mu.Lock()
 	defer irc.mu.Unlock()
```

Comment:
IIRC we needed this lock to protect concurrent access (key lookup) to the CacheMap. Now that we don't have that any more, do we still need this?

Reply:
I think we still need to protect the initialization of the cache to avoid running into race condition